### PR TITLE
chore(generator): enable clang-tidy for integration tests

### LINKED ISSUE #5854: Enable clang-tidy for generator integration tests
The integration tests disable `clang-tidy` for both generated and non-generated code:

https://github.com/googleapis/google-cloud-cpp/blob/d756b07aa23e608639bbfea47e122b71c49a40e6/generator/integration_tests/CMakeLists.txt#L30-L31

The motivation is clear from the comments, we don't want to `clang-tidy` the code generated by protobuf or gRPC, but effectively this disables `clang-tidy` in our own code too, e.g., in the tests themselves.  I will implement a more surgical fix to disable clang-tidy, and leave a `TODO` to cleanup all our code.

### PR DESCRIPTION
**Pull request by scotthart**
fixes #5854

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092)

### CODE REVIEW

File: generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc [hunk 1]
```diff
@@ -21,8 +21,7 @@
 
 using ::google::cloud::testing_util::IsOk;
 using ::google::cloud::testing_util::StatusIs;
-using ::testing::_;
-using ::testing::ByMove;
+using ::testing::_;  // NOLINT(bugprone-reserved-identifier)
```
Comment (devbww) [inline generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc:24]: I wonder why we don't need the NOLINT on other such incantations.

Comment (coryan) [reply generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc:24]: This one is in the global namespace, most others are in `google::cloud::*::` and then some anonymous namespace.

Comment (coryan) [reply generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc:24]: Consider moving all these using declarations line 32 or so.

File: generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc [hunk 2]
```diff
@@ -153,7 +153,7 @@ TEST_F(LoggingDecoratorTest, ListLogsError) {
 }
 
 TEST_F(LoggingDecoratorTest, TailLogEntriesRpcNoRpcStreams) {
-  auto mock_response = new MockTailLogEntriesStreamingReadRpc;
+  auto* mock_response = new MockTailLogEntriesStreamingReadRpc;
```
Comment (coryan) [inline generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc:156]: Consider moving this to a `std::unique_ptr<>` instead.

File: generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc [hunk 3]
```diff
@@ -173,7 +173,7 @@ TEST_F(LoggingDecoratorTest, TailLogEntriesRpcNoRpcStreams) {
 }
 
 TEST_F(LoggingDecoratorTest, TailLogEntriesRpcWithRpcStreams) {
-  auto mock_response = new MockTailLogEntriesStreamingReadRpc;
+  auto* mock_response = new MockTailLogEntriesStreamingReadRpc;
```
Comment (coryan) [inline generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc:176]: Ditto, consider using `absl::make_unique<>`.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc, line 24 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092#-MfswAN5_qgsmB9z7wyo-r1-24:-Mft0P4n0tOMhgyN96rI:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c407c73f85b582160d31478652f67e6019e51470/generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc#L24)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Consider moving all these using declarations line 32 or so.
</blockquote></details>

Done.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc, line 24 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092#-MfswAN5_qgsmB9z7wyo-r1-24:-Mft22HN373EYpUZqy7p:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c407c73f85b582160d31478652f67e6019e51470/generator/integration_tests/golden/tests/golden_kitchen_sink_stub_test.cc#L24)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

**Review by coryan — APPROVED** [review]
Do you know why there are raw `new Foo` calls needed?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092)* status: 0 of 6 files reviewed, 3 unresolved discussions (waiting on @coryan and @devbww)

___
*[generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc, line 156 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092#-MfswAN5_qgsmB9z7wyn-r1-156:-MftBIJEE-EXOdvpgtcJ:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c407c73f85b582160d31478652f67e6019e51470/generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc#L156)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Consider moving this to a `std::unique_ptr<>` instead.
</blockquote></details>

Done.
___
*[generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc, line 176 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7092#-MfswAN5_qgsmB9z7wyn-r1-176:-MftBJ-eAUijAdWMdEIn:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c407c73f85b582160d31478652f67e6019e51470/generator/integration_tests/golden/tests/golden_kitchen_sink_logging_decorator_test.cc#L176)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Ditto, consider using `absl::make_unique<>`.
</blockquote></details>

Done.

**Pull request merged**